fuel-txpool/src/interface.rs

```diff
@@ -0,0 +1,192 @@
+use crate::{types::*, Config, TxPool as TxPoolImpl};
+use fuel_core_interfaces::block_importer::ImportBlockBroadcast;
+use fuel_core_interfaces::model::{ArcTx, TxInfo};
+use fuel_core_interfaces::txpool::{Error, TxPoolDb, TxPoolMpsc, TxStatus, TxStatusBroadcast};
+use std::cmp::Reverse;
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::{broadcast, mpsc, RwLock};
+
+/// Acts an interface between Service and Txpool
+pub struct Interface {
+    txpool: RwLock<TxPoolImpl>,
+    broadcast: broadcast::Sender<TxStatusBroadcast>,
+    db: Box<dyn TxPoolDb>,
+}
+
+impl Interface {
+    pub async fn run(
```

Comment:
Would it be better that `run_interface()` is either a standalone function that takes `interface: Arc<Interface>` or `Interface` method that takes `self: Arc<Self>`?
Currently it's an associated function with a bit confusing API: `Interface::run(Arc::new(Interface::new()))`

Reply:
I agree, left it as run but made first input argument as `self: Arc<Self>`